File: CS-Examples/12_Formulas/ReadFormulas.cs
using System;
using System.Data.OleDb;
using System.Drawing;
using System.Collections;
using System.ComponentModel;
using System.Windows.Forms;
using System.Data;
using Spire.Xls;

namespace ReadFormulas
{
	public partial class Form1 : Form
    {
        public Form1()
        {
            InitializeComponent();
        }
		private void btnRun_Click(object sender, System.EventArgs e)
		{
            // Create a workbook
			Workbook workbook = new Workbook();
            // Load an existing workbook from a file
            workbook.LoadFromFile(@"..\..\..\..\..\..\Data\ReadFormulas.xlsx");

            // Get the first worksheet
            Worksheet sheet = workbook.Worksheets[0];

            // Get the formula from cell C14
            string formula = sheet.Range["C14"].Formula;

            // Get the numeric value resulting from the formula in cell C14
            string formulaNumberValue = sheet.Range["C14"].FormulaNumberValue.ToString();

           // Show the formula and its numeric value
            textBox1.Text = sheet.Range["C14"].Formula;
			textBox2.Text = sheet.Range["C14"].FormulaNumberValue.ToString();

            // Dispose of the workbook object to release resources
            workbook.Dispose();
        }
		private void button1_Click(object sender, System.EventArgs e)
		{
            this.ExcelDocViewer(@"..\..\..\..\..\..\Data\ReadFormulas.xlsx");
		}

		private void ExcelDocViewer( string fileName )
		{
			try
			{
				System.Diagnostics.Process.Start(fileName);
			}
			catch{}
		}

        private void btnClose_Click(object sender, EventArgs e)
        {
            Close();
        }


	}
}

File: CS-Examples/12_Formulas/ReadFormulas.vb
Imports System.Data.OleDb
Imports System.Collections
Imports System.ComponentModel
Imports Spire.Xls

Namespace ReadFormulas
	Partial Public Class Form1
		Inherits Form
		Public Sub New()
			InitializeComponent()
		End Sub
		Private Sub btnRun_Click(ByVal sender As Object, ByVal e As EventArgs) Handles btnRun.Click
            ' Create a new Workbook object
            Dim workbook As New Workbook()

            ' Load the workbook from the specified file path
            workbook.LoadFromFile("..\..\..\..\..\..\Data\ReadFormulas.xlsx")

            ' Get the first worksheet from the workbook
            Dim sheet As Worksheet = workbook.Worksheets(0)

            ' Assign the formula of cell C14 to the text property of textBox1
            textBox1.Text = sheet.Range("C14").Formula
            ' Convert the numeric value of the formula in cell C14 to a string and assign it to the text property of textBox2
            textBox2.Text = sheet.Range("C14").FormulaNumberValue.ToString()
		End Sub
		Private Sub button1_Click(ByVal sender As Object, ByVal e As EventArgs) Handles button1.Click
			Me.ExcelDocViewer("..\..\..\..\..\..\Data\ReadFormulas.xlsx")
		End Sub

		Private Sub ExcelDocViewer(ByVal fileName As String)
			Try
				Process.Start(fileName)
			Catch
			End Try
		End Sub

		Private Sub btnClose_Click(ByVal sender As Object, ByVal e As EventArgs) Handles btnClose.Click
			Close()
		End Sub


	End Class
End Namespace
